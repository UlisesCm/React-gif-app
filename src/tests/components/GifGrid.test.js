import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";

jest.mock("../../hooks/useFetchGifs");

describe("Test del componente <GifGrid/>", () => {
  const category = "Cualquier Categoria";

  test("compara el Snapshot", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: false,
    });

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de mostrar items cuando se cargan imagenes useGetchGifs", () => {
    const gifs = [
      {
        id: "ABD",
        url: "http://localhost/cualquiercosa.jps",
        title: "Cualquiercosa",
      },
      {
        id: "123",
        url: "http://localhost/cualquiercosa.jps",
        title: "Cualquiercosa",
      },
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
  });
});
