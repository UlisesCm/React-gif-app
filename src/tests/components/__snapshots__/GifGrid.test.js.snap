// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test del componente <GifGrid/> compara el Snapshot 1`] = `ShallowWrapper {}`;

exports[`Test del componente <GifGrid/> debe de mostrar items cuando se cargan imagenes useGetchGifs 1`] = `ShallowWrapper {}`;
